R/lazy-select-query.R

```diff
@@ -19,7 +19,10 @@ lazy_select_query <- function(x,
   # stopifnot(is.character(group_by))
   stopifnot(is_lazy_sql_part(order_by))
   check_number_whole(limit, allow_infinite = TRUE, allow_null = TRUE)
-  check_bool(distinct)
+  # distinct = FALSE     -> no distinct
+  # distinct = TRUE      -> distinct over all columns
+  # distinct = columns   -> DISTINCT ON (...)
+  stopifnot(is.logical(distinct) || is_lazy_sql_part(distinct))
```

Comment:
```suggestion
  stopifnot(is_bool(distinct) || is_lazy_sql_part(distinct))
```

Since `is.logical()` allows vectors and `NA`, neither of which we support.